### PR TITLE
Adding tags in instapaper import

### PR DESCRIPTION
**Pull request by itayfo**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    |no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | no
| Translation   | no
| Fixed tickets | #...
| License       | MIT

### CODE REVIEW

File: src/Wallabag/ImportBundle/Import/InstapaperImport.php [hunk 1]
```diff
@@ -115,6 +117,22 @@ public function parseEntry(array $importedEntry)
         $entry->setUrl($importedEntry['url']);
         $entry->setTitle($importedEntry['title']);
 
+        if($importedEntry['tags']) {
+            foreach(explode(',', $importedEntry['tags']) as $tag) {
+                if(strlen($tag) === 0)continue;
+                $existingTag = $this->em
+                    ->getRepository('WallabagCoreBundle:Tag')
+                    ->findOneByLabel($tag);
```
Comment (nicosomb) [inline src/Wallabag/ImportBundle/Import/InstapaperImport.php]: There is a performance issue with this method, see https://github.com/wallabag/wallabag/issues/2864. 
We need to work on it.

**Comment by nicosomb** [comment]
@itayfo: how did you manage to add multiple tags (folders) on a URL? With the Instapaper web application, it's not possible. 

Moreover, you defined `$data[4]` for tags, but here is a sample line from my own export: 
`http://www.lemonde.fr/les-decodeurs/article/2017/01/17/comment-benoit-hamon-a-rabote-son-revenu-universel-en-pleine-campagne_5064235_4355770.html,"En pleine campagne, Benoît Hamon rabote (puis remet) son « revenu universel »",,plop` where `plop` is a folder I created. So, it's the 3rd element, not the 4th. 

Finally, I have my other lines with `unread` in this column. So with your pull request, we'll have `Unread` as a tag.

File: src/Wallabag/ImportBundle/Import/InstapaperImport.php [hunk 2]
```diff
@@ -115,6 +117,24 @@ public function parseEntry(array $importedEntry)
         $entry->setUrl($importedEntry['url']);
         $entry->setTitle($importedEntry['title']);
 
+        if ($importedEntry['tags']) {
```
Comment (j0k3r) [inline src/Wallabag/ImportBundle/Import/InstapaperImport.php]: Might be better to re-use `assignTagsToEntry`

```php
        if (!empty($data['tags'])) {
            $this->contentProxy->assignTagsToEntry(
                $entry,
                $data['tags'],
                $this->em->getUnitOfWork()->getScheduledEntityInsertions()
            );
        }
```

**Comment by itayfo** [comment]
@nicosomb 

Apologizes. It is not a valid pull request. 
I wanted to quickly support mass import of URLS into Wallabag.
I believe The proper way to do is create some 'custom' importer, which I did not. 

We used this code as a part of internal importer.
Info was taken from RSS feeds.

**Comment by nicosomb** [comment]
Can I close this PR?

**Comment by j0k3r** [comment]
But adding tag from instapaper import isn't a bad idea, nope? :thinking:

**Comment by nicosomb** [comment]
Yeah, sorry, I replied too quickly.

**Comment by itayfo** [comment]
Yes, you can. 

But I do think we need to support some custom import type, that could importing tags as well.

**Comment by j0k3r** [comment]
Since you removed your repository I can no longer work on what you already did.
So I'll close that PR in favor of a new one I just created https://github.com/wallabag/wallabag/pull/3168

Thanks for the idea!

**Pull request closed without merging**